### ISSUE TITLE
some question about mmseg-related Dataloader

### ISSUE THREAD

**Issue by endaoguansanlu** [open-mmlab/mmcv#2376]
Hello mmcv developers! I used a loss module in the process of using mmsegmentation, but this module is very slow to load in a single GPU, so I want to load it in the Dataloader, but I don't know how to modify the mmseg-related Dataloader in mmcv. So I hope you developers can give me a little guidance, thank you!

Below is a demo of the module I want to use loaded in Dataloader.

`
import torch
from torch.utils.data import Dataset, DataLoader
import numpy as np
from scipy.ndimage import distance_transform_edt as distance
from torchvision import transforms
from functools import partial
from operator import itemgetter
from tqdm import tqdm
import cv2

def kl_div(a,b): # q,p
    return F.softmax(b, dim=1) * (F.log_softmax(b, dim=1) - F.log_softmax(a, dim=1))   

def one_hot2dist(seg):
    res = np.zeros_like(seg)
    for i in range(len(seg)):
        posmask = seg[i].astype(np.bool)
        if posmask.any():
            negmask = ~posmask
            res[i] = distance(negmask) * negmask - (distance(posmask) - 1) * posmask
    return res

def class2one_hot(seg, C):
    seg = seg.unsqueeze(dim=0) if len(seg.shape) == 2 else seg
    res = torch.stack([seg == c for c in range(C)], dim=1).type(torch.int32)
    return res

class DummyDataset(Dataset):
        def __init__(self):
            self.dist_map_transform = transforms.Compose([
                lambda img: img.unsqueeze(0),
                lambda nd: nd.type(torch.int64),
                partial(class2one_hot, C=1),
                itemgetter(0),
                lambda t: t.cpu().numpy(),
                one_hot2dist,
                lambda nd: torch.tensor(nd, dtype=torch.float32)
            ])
        
        def get_dist_maps(self, target):
            target_detach = target.clone().detach()
            dist_maps = torch.cat([self.dist_map_transform(target_detach[i]) for i in range(target_detach.shape[0])])
            out = -dist_maps
            out = torch.where(out>0, out, torch.zeros_like(out))
            return out

        def __len__(self):
                return 100

        def __getitem__(self, index):
                dummy_gt_np = np.random.randint(-10,2,[5,20,20]) # random, onehot, [gt_class, h, w]
                dummy_gt_np[dummy_gt_np<0] = 0 # make it sparse
                dummy_gt = torch.from_numpy(dummy_gt_np)
                dist = self.get_dist_maps(dummy_gt)
                return dummy_gt_np, dist

if __name__ == '__main__':
    dummy_dataset = DummyDataset()
    dummy_loader = DataLoader(dummy_dataset, batch_size=2)
    for i, (gt, dist) in enumerate(tqdm(dummy_loader)):
        cv2.imwrite(f'vis/{i}.png', gt[0][0].numpy().astype(np.uint8)*255)
        cv2.imwrite(f'vis/{i}_dist.png', dist[0][0].numpy().astype(np.uint8) * 10)
        pass
`
